atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTable.java

```diff
@@ -223,7 +237,31 @@ private static void executeIgnoringTableMappingNotFound(
         }
     }
 
-    private void clearTableSizeCacheAndDropTableMetadata() {
+    private void clearTableSizeCacheAndDropTableMetadataCaseInsensitive() {
+        valueStyleCache.clearCacheForTable(tableRef);
+        long numberOfMatchingTableReferences = conns.get()
+                .selectCount(
+                        config.metadataTable().getQualifiedName(),
+                        "LOWER(table_name) = LOWER(?)",
+                        tableRef.getQualifiedName());
+
+        Preconditions.checkState(
+                numberOfMatchingTableReferences == 1,
+                "There are multiple tables that have the same case"
+                        + " insensitive table reference. Throwing to avoid accidentally deleting the wrong table reference."
+                        + " Please contact support to delete the metadata, which will involve deleting the row from"
+                        + " the DB manually.",
+                SafeArg.of("numberOfMatchingTableReferences", numberOfMatchingTableReferences),
+                UnsafeArg.of("tableReference", tableRef));
+
+        conns.get()
+                .executeUnregisteredQuery(
+                        "DELETE FROM " + config.metadataTable().getQualifiedName() + " WHERE LOWER(table_name) ="
+                                + " LOWER(?)",
+                        tableRef.getQualifiedName());
```

Comment:
I'd leave a comment here (probably just above this line) that there is the obvious race condition, but we are explicitly saying that we don't support that case.